MediaManager/Platforms/Android/MediaSession/MediaSessionCallback.cs

```diff
@@ -44,31 +46,191 @@ protected MediaSessionCallback(IntPtr javaReference, JniHandleOwnership transfer
-        public override void OnPlayFromSearch(string query, Bundle extras) => OnPlayFromSearchImpl?.Invoke(query, extras);
-        public override void OnPlayFromUri(global::Android.Net.Uri uri, Bundle extras) => OnPlayFromUriImpl?.Invoke(uri, extras);
-        public override void OnPrepare() => OnPrepareImpl?.Invoke();
-        public override void OnPrepareFromMediaId(string mediaId, Bundle extras) => OnPrepareFromMediaIdImpl?.Invoke(mediaId, extras);
-        public override void OnPrepareFromSearch(string query, Bundle extras) => OnPrepareFromSearchImpl?.Invoke(query, extras);
-        public override void OnPrepareFromUri(global::Android.Net.Uri uri, Bundle extras) => OnPrepareFromUriImpl?.Invoke(uri, extras);
-        public override void OnRemoveQueueItem(MediaDescriptionCompat description) => OnRemoveQueueItemImpl?.Invoke(description);
-        public override void OnRemoveQueueItemAt(int index) => OnRemoveQueueItemAtImpl?.Invoke(index);
-        public override void OnRewind() => OnRewindImpl?.Invoke();
-        public override void OnSeekTo(long pos) => OnSeekToImpl?.Invoke();
-        public override void OnSetCaptioningEnabled(bool enabled) => OnSetCaptioningEnabledImpl?.Invoke();
-        public override void OnSetRating(RatingCompat rating) => OnSetRatingImpl?.Invoke(rating);
-        public override void OnSetRating(RatingCompat rating, Bundle extras) => OnSetRatingWithExtrasImpl?.Invoke(rating, extras);
-        public override void OnSetRepeatMode(int repeatMode) => OnSetRepeatModeImpl?.Invoke();
-        public override void OnSetShuffleMode(int shuffleMode) => OnSetShuffleModeImpl?.Invoke();
-        public override void OnSkipToNext() => OnSkipToNextImpl?.Invoke();
-        public override void OnSkipToPrevious() => OnSkipToPreviousImpl?.Invoke();
-        public override void OnSkipToQueueItem(long id) => OnSkipToQueueItemImpl?.Invoke();
-        public override void OnStop() => OnStopImpl?.Invoke();
+        public override void OnAddQueueItem(MediaDescriptionCompat description)
+        {
+            OnAddQueueItemImpl?.Invoke(description);
+        }
+
+        public override void OnAddQueueItem(MediaDescriptionCompat description, int index)
+        {
+            OnAddQueueItemAtIndexImpl?.Invoke(description, index);
+        }
+
+        public override void OnFastForward()
+        {
+            if (MediaManager.BluetoothControls.SeekForwardImpl != null)
+            {
+                MediaManager.BluetoothControls.SeekForwardImpl.Invoke();
+            }
+            else if (MediaManager.BluetoothControls.SkipForwardImpl != null)
+            {
+                MediaManager.BluetoothControls.SkipBackwardImpl.Invoke();
+            }
+            else
+            {
+                OnFastForwardImpl?.Invoke();
+            }
+        }
+
+        public override void OnRewind()
+        {
+            if (MediaManager.BluetoothControls.SeekBackwardImpl != null)
```

Comment:
So, there may be times that we want to have SeekBackward and SkipBackward fire separate events instead of it being an either/or situation